components/datetime/src/pattern/error.rs

```diff
@@ -0,0 +1,17 @@
+use crate::fields;
+
+#[derive(Debug)]
+pub enum Error {
+    Unknown,
+    FieldTooLong,
```

Comment:
Optional: Remember which field is too long.

Reply:
Not easily, because I parse `usize` to it and would have switch API to pass reference to field type to the parser. Is it ok if I keep it? If we identify a need we can revisit.

Reply:
You might just have to replace a few call sites of `?` to `map_err(|e| Error::FieldTooLong(b))?`.  The reason is that it might be helpful to the user if they got more information on why the error occurred.  But it's not a big deal.

Reply:
Okay. Went a bit deeper and separated different classes of field errors because I think as we get to skeletons fields will become bigger and its ok to split them already. It also made error conversions much easier since symbol errors and length errors don't overlap.